Submitted By: Tushar Teredesai <tushar@linuxfromscratch.org>
Date: 2006-01-17
Initial Package Version: 1.4-p6
Origin: Fedora
Upstream Status: Not submitted, old version.
Description: Misc fixes from Fedora
diff -Naur automake-1.4-p6.orig/automake.in automake-1.4-p6/automake.in
--- automake-1.4-p6.orig/automake.in	2002-07-27 15:55:58.000000000 -0500
+++ automake-1.4-p6/automake.in	2005-11-08 09:38:57.960557296 -0600
@@ -288,7 +288,7 @@
      'fp_FUNC_FNMATCH', "use \`AC_FUNC_FNMATCH'",
      'AM_SANITY_CHECK_CC', "automatically done by \`AC_PROG_CC'",
      'AM_PROG_INSTALL', "use \`AC_PROG_INSTALL'",
-     'AM_EXEEEXT', "use \`AC_EXEEXT'",
+     'AM_EXEEXT', "use \`AC_EXEEXT'",
      'AM_CYGWIN32', "use \`AC_CYGWIN32'",
      'AM_MINGW32', "use \`AC_MINGW32'",
 
@@ -1126,9 +1126,11 @@
 	    # the directory.  Later.
 	    if (/\//)
 	    {
-		&am_error
-		    ("not supported: source file `$_' is in subdirectory");
-		next;
+		# It breaks gas/Makefile.in in binutils. For now, we
+		# turn off this warning.
+		# &am_error
+		# ("not supported: source file `$_' is in subdirectory");
+		# next;
 	    }
 
 	    # Split file name into base and extension.
@@ -5288,8 +5290,7 @@
 sub quote_cond_val
 {
     local ($val) = @_;
-    $val =~ s/ /\001/g;
-    $val =~ s/\t/\003/g;
+    $val =~ tr/ \t\n/\001\003\004/;
     $val = "\002" if $val eq '';
     return $val;
 }
@@ -5298,8 +5299,7 @@
 sub unquote_cond_val
 {
     local ($val) = @_;
-    $val =~ s/\001/ /g;
-    $val =~ s/\003/\t/g;
+    $val =~ tr/\001\003\004/ \t\n/;
     $val =~ s/\002//g;
     return $val;
 }
@@ -5606,6 +5606,9 @@
     local ($var, $val, $cond) = @_;
     local (@result);
 
+    # Strip backslashes
+    $val =~ s/\\(\n|$)/ /g;
+
     foreach (split (' ', $val))
     {
 	# If a comment seen, just leave.
@@ -5912,10 +5915,6 @@
 	    else
 	    {
 		$saw_bk = /\\$/;
-		# Chop newline and backslash if this line is
-		# continued.  ensure trailing whitespace exists.
-		chop if $saw_bk;
-		chop if $saw_bk;
 		$contents{$last_var_name} .= ' '
 		    unless $contents{$last_var_name} =~ /\s$/;
 		$contents{$last_var_name} .= $_;
@@ -6026,7 +6025,7 @@
 	    local ($value);
 	    if ($3 ne '' && substr ($3, -1) eq "\\")
 	    {
-		$value = substr ($3, 0, length ($3) - 1);
+		$value = $3 . "\n";
 	    }
 	    else
 	    {
@@ -6040,6 +6039,12 @@
 		{
 		    $contents{$last_var_name} = '@' . $last_var_name . '@';
 		}
+		if (substr ($contents{$last_var_name}, -1) eq "\n")
+		{
+		    # Insert a backslash before a trailing newline.
+		    $contents{$last_var_name}
+			= substr ($contents{$last_var_name}, 0, -1) . "\\\n";
+		}
 		$contents{$last_var_name} .= ' ' . $value;
 	    }
 	    else
@@ -6242,9 +6247,15 @@
 	    {
 		local ($vcond) = shift (@cond_vals);
 		local ($val) = &unquote_cond_val (shift (@cond_vals));
-		$output_vars .= ($vcond . $curs
-				 . ' ' . $def_type{$curs} . '= '
-				 . $val . "\n");
+		$output_vars .= ($vcond . $curs . ' '
+				 . $def_type{$curs} . "= ");
+		local ($line);
+		foreach $line (split ("\n", $val))
+		{
+		    $output_vars .= $vcond . $line . "\n";
+		}
+		$output_vars .= "\n"
+		    if $val eq '';
 	    }
 	}
 	else
@@ -6977,9 +6988,11 @@
 		    $suppress = 1;
 
 		    # Maybe run libtoolize.
+		    @libtoolize_args = ('--automake');
+		    push @libtoolize_args, '-c'  if $copy_missing;
 		    if ($seen_libtool
 			&& grep ($_ eq $file, @libtoolize_files)
-			&& system ('libtoolize', '--automake'))
+			&& system ('libtoolize', @libtoolize_args))
 		    {
 			$message = "installing \`$errfile'";
 			$suppress = 0;
diff -Naur automake-1.4-p6.orig/subdirs.am automake-1.4-p6/subdirs.am
--- automake-1.4-p6.orig/subdirs.am	2001-07-17 00:47:36.000000000 -0500
+++ automake-1.4-p6/subdirs.am	2005-11-08 09:40:07.333011080 -0600
@@ -27,7 +27,7 @@
 all-recursive install-data-recursive install-exec-recursive \
 installdirs-recursive install-recursive uninstall-recursive @INSTALLINFO@ \
 check-recursive installcheck-recursive info-recursive dvi-recursive:
-	@set fnord $(MAKEFLAGS); amf=$$2; \
+	@set fnord $$MAKEFLAGS; amf=$$2; \
 	dot_seen=no; \
 	target=`echo $@ | sed s/-recursive//`; \
 	list='$(SUBDIRS)'; for subdir in $$list; do \
@@ -55,7 +55,7 @@
 ## bombs.
 mostlyclean-recursive clean-recursive distclean-recursive \
 maintainer-clean-recursive:
-	@set fnord $(MAKEFLAGS); amf=$$2; \
+	@set fnord $$MAKEFLAGS; amf=$$2; \
 	dot_seen=no; \
 	rev=''; list='$(SUBDIRS)'; for subdir in $$list; do \
 	  rev="$$subdir $$rev"; \
